inst/include/Rcpp/platform/compiler.h

```diff
@@ -24,173 +22,35 @@
 
 // NB: A vast list valid identifiers is at these wiki pages:
 //     http://sourceforge.net/p/predef/wiki/Home/
-
-#undef GOOD_COMPILER_FOR_RCPP
-#ifdef __GNUC__
-#define GOOD_COMPILER_FOR_RCPP
-#endif
-#ifdef __SUNPRO_CC
-#define GOOD_COMPILER_FOR_RCPP
-#endif
-#ifdef __clang__
-#define GOOD_COMPILER_FOR_RCPP
-#endif
-#ifdef __INTEL_COMPILER
+#if defined(__GNUC__) || defined(__SUNPRO_CC) || defined(__clang__) || defined(__INTEL_COMPILER)
 #define GOOD_COMPILER_FOR_RCPP
+#else
+#error "This compiler is not supported"
 #endif
 
-#ifndef GOOD_COMPILER_FOR_RCPP
-# error "This compiler is not supported"
+// New simpler test and minimal standard: C++11 or else we die
+#if __cplusplus < 201103L
+#error "The C++ compilation standard is too old: use C++11 or newer."
 #endif
 
 #ifdef __GNUC__
     #define GCC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
-    // g++ 4.5 does not seem to like some of the fast indexing
-    #if GCC_VERSION >= 40500
-        #define IS_GCC_450_OR_LATER
-    #endif
-    // g++ 4.6 switches from exception_defines.h to bits/exception_defines.h
-    #if GCC_VERSION < 40600
-        #define IS_EARLIER_THAN_GCC_460
-    #endif
-    #if GCC_VERSION >= 40600
-        #define IS_GCC_460_OR_LATER
-    #endif
 #endif
 
-// Check for the presence of C++0x (or later) support
-#if defined(__GXX_EXPERIMENTAL_CXX0X__) || (__cplusplus >= 201103L)
-  #define RCPP_USING_CXX0X_OR_LATER
-#endif
-
-// Check C++0x/11 features
-#if defined(__INTEL_COMPILER)
-    #if __cplusplus >= 201103L
-        #define RCPP_USING_CXX11
-        #if __INTEL_COMPILER >= 1210
-            #define HAS_VARIADIC_TEMPLATES
-        #endif
-        #if __INTEL_COMPILER >= 1100
-            #define HAS_STATIC_ASSERT
-        #endif
-    #endif
-#elif defined(__clang__)
-    #if __cplusplus >= 201103L
-        #define RCPP_USING_CXX11
-        #if __has_feature(cxx_variadic_templates)
-            #define HAS_VARIADIC_TEMPLATES
-        #endif
-        #if __has_feature(cxx_static_assert)
-            #define HAS_STATIC_ASSERT
-        #endif
-    #endif
-#elif defined(__GNUC__)
-    #ifdef __GXX_EXPERIMENTAL_CXX0X__
-        #if GCC_VERSION >= 40300
-            #define HAS_VARIADIC_TEMPLATES
-            #define HAS_STATIC_ASSERT
-        #endif
-    #endif
-	#if GCC_VERSION >= 40800 && __cplusplus >= 201103L
-		#define RCPP_USING_CXX11
-	#endif
-#endif
-
-// Check C++0x headers
+// C++11 features -- that used to be carefully tested for or worked around via CXX0X / TR1
+#define RCPP_USING_CXX11
+#define HAS_VARIADIC_TEMPLATES
```

Comment:
Nobody on CRAN uses HAS_VARIADIC_TEMPLATES, and only RInside uses RCPP_USING_CXX11.

Reply:
Even if we keep these, I would remove all checks for these two throughout our code. There is quite a number of instances.

Reply:
Well **of course!**  But because there are so many I plan to **very carefully** remove them one by one.

Reply:
(And 'so many' I meant uses of 'has_variadic_templates' as our main C++11 bonus.  'rcpp_using_cxx11' is another.  Should be fixable over two PRs maybe over two weeks.  It's looking good and went well so far so fingers crossed!)